jscomp/others/belt_Array.mli

```diff
@@ -522,6 +633,23 @@ val reduceWithIndex:  'a array -> 'b -> ('b -> 'a -> int -> 'b) -> 'b
     ]}
 *)
 
+val joinWith: string array -> string -> string
+(** [joinWith xs sep]
+    
```

Comment:
this API looks too specific, could we make it like this:
joinWithU: 'a array ->sep:string ->  ('a -> string [@bs]) -> string